resources/testharness.js

```diff
@@ -1258,6 +1265,19 @@ policies and contribution forms [3].
                 throw e;
             }
 
+            if ((typeof IdlHarnessError !== 'undefined')
```

Comment:
The way that `assert_throws` is used for the per-spec tests themselves is to just check that the right kind of exception is thrown, as well as it can be checked (see "but we can't" below) and to never assert anything about the exception message itself, since the messages aren't defined in web specs and are UA-defined.

For idlharness.js it is kinda useful to check that the right exception message has been thrown, but I wonder if it isn't better to use a custom assertion for those tests?

An alternative could be to allow messages to be asserted more generically, adding it to the `if (typeof code === "object")` branch below, that if `code.message` is a non-empty string, we assert that the throw exception matches. That would remove the need to have a special `IdlHarnessError`.

WDYT?

Reply:
I see there's also an AssertionError in testharness.js that we could perhaps use. It's immediately rethrown by assert_throws. It really depend on what problem https://github.com/w3c/web-platform-tests/issues/10162 started out with, if it's just about fixing the inconsistency or if there's also some test that needs to be fixed.

Reply:
To see if that idea is at all viable I sent https://chromium-review.googlesource.com/c/chromium/src/+/980757 to see if there are lots of existing tests that pass a message that *shouln't* be checked.

Reply:
Yep, there were a handful, like this one:
https://chromium.googlesource.com/chromium/src/+/15e299fd19da9cf6f74dbcb78b6bc4865f7a5a35/third_party/WebKit/LayoutTests/external/wpt/mediasession/setactionhandler.html

Seems manageable if we want to do it that way, though.

Reply:
I agree that it's bad to reference `IdlHarnessError` from here.